crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/RepOrderService.java

```diff
@@ -122,4 +129,95 @@ public ReviewResult getReviewResult(ApiIOJAppeal apiIOJAppeal) {
+        /*
+            PROCEDURE determine_crown_rep_type (p_app_obj    IN OUT    application_type) IS BEGIN -- determine_crown_rep_type
+       if p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_reporder_decision is not null
+       THEN
+       */
+        if (crownCourtSummary.getRepOrderDecision() != null) {
+            /*
+              IF p_app_obj.mags_outcome_object.outcome IN ('SENT FOR TRIAL','COMMITTED FOR TRIAL')
+              THEN
+            */
+            if (requestDTO.getMagCourtOutcome() == MagCourtOutcome.SENT_FOR_TRIAL
+                    || requestDTO.getMagCourtOutcome() == MagCourtOutcome.COMMITTED_FOR_TRIAL) {
+                determineRepTypeByDecisionReason(requestDTO, crownCourtSummary);
+            }
+            determineRepTypeByRepOrderDecision(requestDTO, crownCourtSummary);
+            /*
+              IF p_app_obj.case_type_object.case_type = 'CC ALREADY'
+              THEN
+                 p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_REP_TYPE := 'Crown Court Only';
+              END IF;
+            */
+            if (requestDTO.getCaseType() == CaseType.CC_ALREADY) {
+                crownCourtSummary.setRepType(CROWN_COURT_ONLY);
+            }
+            /*
+              IF p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_WITHDRAWAL_DATE is not null
+              THEN
+                 p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_REP_TYPE := 'Declined Rep Order';
+              END IF;
+            */
+            if (crownCourtSummary.getWithdrawalDate() != null) {
+                crownCourtSummary.setRepType(DECLINED_REP_ORDER);
+            }
+        }
+        return crownCourtSummary;
+    }
+
+    public void determineRepTypeByRepOrderDecision(CrownCourtsActionsRequestDTO requestDTO, ApiCrownCourtSummary crownCourtSummary) {
+        /*
+          IF p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_reporder_decision like 'Granted%'
+           THEN
+             IF (p_app_obj.case_type_object.case_type = 'APPEAL CC')
+             OR (p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_reporder_decision like 'Granted%Pass%'
+                AND p_app_obj.case_type_object.case_type = 'COMMITAL')
+             THEN
+            p_app_obj.crown_court_overview_object.crown_court_summary_object.CC_REP_TYPE := 'Crown Court Only';
+            END IF;
+          */
+        if ((GRANTED_FAILED_MEANS_TEST.equals(crownCourtSummary.getRepOrderDecision()) && requestDTO.getCaseType() == CaseType.APPEAL_CC) ||
+                ((GRANTED_PASSPORTED.equals(crownCourtSummary.getRepOrderDecision()) || GRANTED_PASSED_MEANS_TEST.equals(crownCourtSummary.getRepOrderDecision()))
+                        && (requestDTO.getCaseType() == CaseType.APPEAL_CC || requestDTO.getCaseType() == CaseType.COMMITAL))) {
+            crownCourtSummary.setRepType(CROWN_COURT_ONLY);
```

Comment:
I think this could probably be simplified, how about something like this?

```
List<CaseType> caseTypes = List.of(CaseType.APPEAL_CC, CaseType.COMMITAL);
List<String> repOrderDecisions = List.of(GRANTED_FAILED_MEANS_TEST, GRANTED_PASSED_MEANS_TEST, GRANTED_PASSPORTED);
        
if (repOrderDecisions.contains(crownCourtSummary.getRepOrderDecision()) && caseTypes.contains(requestDTO.getCaseType())) {
    crownCourtSummary.setRepType(CROWN_COURT_ONLY);
}
```

Reply:
Done.